Author: Christophe Dumez <chris@qbittorrent.org>
Description: Fix possible crash when using alternative speed limits
Origin: upstream, http://qbittorrent.svn.sourceforge.net/viewvc/qbittorrent?view=revision&revision=4186

--- a/src/bittorrent.cpp
+++ b/src/bittorrent.cpp
@@ -637,8 +637,20 @@
     s->set_download_rate_limit(Preferences::getAltGlobalDownloadLimit()*1024);
     s->set_upload_rate_limit(Preferences::getAltGlobalUploadLimit()*1024);
   } else {
-    s->set_download_rate_limit(Preferences::getGlobalDownloadLimit()*1024);
-    s->set_upload_rate_limit(Preferences::getGlobalUploadLimit()*1024);
+    int down_limit = Preferences::getGlobalDownloadLimit();
+    if(down_limit <= 0) {
+      down_limit = -1;
+    } else {
+      down_limit *= 1024;
+    }
+    s->set_download_rate_limit(down_limit);
+    int up_limit = Preferences::getGlobalUploadLimit();
+    if(up_limit <= 0) {
+      up_limit = -1;
+    } else {
+      up_limit *= 1024;
+    }
+    s->set_upload_rate_limit(up_limit);
   }
   emit alternativeSpeedsModeChanged(alternative);
 }
--- a/src/preferences.h
+++ b/src/preferences.h
@@ -310,7 +310,10 @@
 
   static int getAltGlobalDownloadLimit() {
     QSettings settings("qBittorrent", "qBittorrent");
-    return settings.value(QString::fromUtf8("Preferences/Connection/GlobalDLLimitAlt"), 10).toInt();
+    int ret = settings.value(QString::fromUtf8("Preferences/Connection/GlobalDLLimitAlt"), 10).toInt();
+    if(ret <= 0)
+      ret = 10;
+    return ret;
   }
 
   static void setAltGlobalDownloadLimit(int limit) {
@@ -321,7 +324,10 @@
 
   static int getAltGlobalUploadLimit() {
     QSettings settings("qBittorrent", "qBittorrent");
-    return settings.value(QString::fromUtf8("Preferences/Connection/GlobalUPLimitAlt"), 10).toInt();
+    int ret = settings.value(QString::fromUtf8("Preferences/Connection/GlobalUPLimitAlt"), 10).toInt();
+    if(ret <= 0)
+      ret = 10;
+    return ret;
   }
 
   static void setAltGlobalUploadLimit(int limit) {
