From a4d9e65d386df680930be243d7842e6766756fe9 Mon Sep 17 00:00:00 2001
From: bob23450 <bzanetti00@gmail.com>
Date: Tue, 4 Feb 2014 10:35:43 +0100
Subject: [PATCH] Fix nox build. Closes #1368

---
 src/src.pro | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

Index: qbittorrent/src/src.pro
===================================================================
--- qbittorrent.orig/src/src.pro	2014-02-27 10:53:58.877158849 -0500
+++ qbittorrent/src/src.pro	2014-02-27 10:53:58.869158849 -0500
@@ -107,8 +107,7 @@
            scannedfoldersmodel.h \
            qinisettings.h \
            smtp.h \
-           dnsupdater.h \
-           messageboxraised.h
+           dnsupdater.h
 
 
 SOURCES += main.cpp \
@@ -117,8 +116,7 @@
            misc.cpp \
            fs_utils.cpp \
            smtp.cpp \
-           dnsupdater.cpp \
-           messageboxraised.cpp
+           dnsupdater.cpp
 
 nox {
   HEADERS += headlessloader.h
@@ -152,7 +150,8 @@
               loglistwidget.h \
               addnewtorrentdialog.h \
               autoexpandabledialog.h \
-              statsdialog.h
+              statsdialog.h \
+              messageboxraised.h
 
   SOURCES += mainwindow.cpp \
              ico.cpp \
@@ -171,7 +170,8 @@
              loglistwidget.cpp \
              addnewtorrentdialog.cpp \
              autoexpandabledialog.cpp \
-             statsdialog.cpp
+             statsdialog.cpp \
+             messageboxraised.cpp
 
   win32 {
     HEADERS += programupdater.h
