From: Cristian Greco <cristian@debian.org>
Date: Wed, 15 Feb 2017 20:27:16 +0100
Subject: honor-buildflags

---
 unixconf.pri | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- qbittorrent.orig/unixconf.pri
+++ qbittorrent/unixconf.pri
@@ -15,8 +15,9 @@ lessThan(QT_MAJOR_VERSION, 5): QMAKE_CXX
 # COMPILATION SPECIFIC
 !nogui:dbus: QT += dbus
 
-QMAKE_CXXFLAGS += -Wformat -Wformat-security
-!haiku: QMAKE_LFLAGS_APP += -rdynamic
+QMAKE_CXXFLAGS += -Wformat -Wformat-security $$(CXXFLAGS) $$(CPPFLAGS)
+QMAKE_LFLAGS_RELEASE += $$(LDFLAGS)
+!haiku: QMAKE_LFLAGS_APP += -rdynamic $$(LDFLAGS)
 
 # Man page
 nogui {
