From: Steven Chamberlain <steven@pyro.eu.org>
Date: Sun, 21 Feb 2016 01:30:09 +0000
Subject: use -lexecinfo on FreeBSD, but not GNU/kFreeBSD

---
 configure    | 2 +-
 configure.ac | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 2c72cb2..f72ea6c 100755
--- a/configure
+++ b/configure
@@ -4248,7 +4248,7 @@ fi
 # Detect OS
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether OS is FreeBSD" >&5
 $as_echo_n "checking whether OS is FreeBSD... " >&6; }
-if expr "$host_os" : ".*freebsd.*" > /dev/null; then :
+if expr "$host_os" : "freebsd.*" > /dev/null; then :
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
       LIBS="-lexecinfo $LIBS"
diff --git a/configure.ac b/configure.ac
index c008c57..ab9ff80 100644
--- a/configure.ac
+++ b/configure.ac
@@ -62,7 +62,7 @@ AC_ARG_ENABLE(qt-dbus,
 
 # Detect OS
 AC_MSG_CHECKING([whether OS is FreeBSD])
-AS_IF([expr "$host_os" : ".*freebsd.*" > /dev/null],
+AS_IF([expr "$host_os" : "freebsd.*" > /dev/null],
       [AC_MSG_RESULT([yes])
       LIBS="-lexecinfo $LIBS"],
       [AC_MSG_RESULT([no])])
